### PR TITLE
Expose more `Page` methods

### PR DESCRIPTION
**Pull request by johnvictor1965**
Despite being implemented in `Target`, Puppeteer's `authenticate()` API is not available for regular `Page` users. This PR adds a new target message to port that. Same goes for `opener_id()`.

### CODE REVIEW

**Comment by johnvictor1965** [comment]
Closes #145

File: src/handler/target.rs [hunk 1]
```diff
@@ -599,24 +603,6 @@ pub struct TargetConfig {
...
-    ) -> Self {
-        Self {
-            ignore_https_errors,
-            request_timeout,
-            viewport,
-            request_intercept,
-            cache_enabled,
-        }
-    }
-}
```
Comment (mattsse) [inline src/handler/target.rs:618]: this is unrelated?

**Review by mattsse — APPROVED** [review]
lgtm,
one q re removed targetconfig function, 
I think this is probably unused but removing this would be a breaking change

File: src/handler/target.rs (hunk 1 above)
Comment (johnvictor1965) [reply src/handler/target.rs:618]: It is. My bad. I just suddenly started refactoring your crate at one point.

**Comment by johnvictor1965** [comment]
I just added another one. And renamed `Target::opener()` to `Target::opener_id()`. Would that be a breaking change despite no one actually using `Target` in user code?

**Review by mattsse — APPROVED** [review]
this seems fine, I don't think this is used manually

**Comment by johnvictor1965** [comment]
Lemme just fix those imports for you
